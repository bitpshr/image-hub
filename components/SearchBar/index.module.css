@import "~lib/variables.css";

.control {
	margin-bottom: calc(var(--grid-size) * 2);
}

.search {
	border-radius: calc(var(--grid-size) / 4);
	border: 1px solid var(--input-border);
	font: inherit;
	margin: 0 calc(var(--grid-size) * 2) calc(var(--grid-size) * 2) 0;
	max-width: calc(var(--grid-size) * 75);
	padding: var(--grid-size) calc(var(--grid-size) * 2);
	width: 100%;
}

.uploadControl {
	position: relative;
}

.uploadControl > button {
	z-index: -1;
}

.uploadControl > input {
	cursor: pointer;
	display: block;
	font-size: 0;
	height: 100%;
	left: 0;
	opacity: 0;
	position: absolute;
	top: 0;
	width: 100%;
	z-index: 2;
}

.small .search {
	margin-right: 0;
}

.small .uploadControl > button {
	width: 100%;
}
