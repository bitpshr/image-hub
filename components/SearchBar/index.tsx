import { FunctionComponent, SyntheticEvent, memo } from "react";

import Button from "@app/components/Button";
import Flex from "@app/components/Flex";
import useMedia from "@app/lib/useMedia";

import css from "./index.module.css";

export interface SearchBarProps {
	onChange: (event: SyntheticEvent<HTMLInputElement>) => void;
	onUpload: (event: SyntheticEvent<HTMLInputElement>) => void;
}

const SearchBar: FunctionComponent<SearchBarProps> = memo((props) => {
	const size = useMedia();

	return (
		<Flex
			alignItems={size.large ? "flex-start" : "stretch"}
			className={`${css.control} ${!size.large ? css.small : ""}`}
			direction={size.large ? "row" : "column"}
		>
			<Flex grow={1}>
				<input
					autoCorrect="off"
					className={css.search}
					onChange={props.onChange}
					placeholder="Search documents..."
					spellCheck={false}
					type="search"
				/>
			</Flex>
			<div className={css.uploadControl}>
				<input
					accept="image/x-png,image/jpeg"
					id="upload"
					onChange={props.onUpload}
					type="file"
				/>
				<Button>upload</Button>
			</div>
		</Flex>
	);
});

export default SearchBar;
