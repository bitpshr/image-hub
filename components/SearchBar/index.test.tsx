import React from "react";
import renderer from "react-test-renderer";

import SearchBar from "./";

test("default render", () => {
	const component = renderer.create(
		<SearchBar onChange={jest.fn} onUpload={jest.fn()} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
