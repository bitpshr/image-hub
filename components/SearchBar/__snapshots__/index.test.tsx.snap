// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<div
  className="control small"
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexGrow": 1,
      }
    }
  >
    <input
      autoCorrect="off"
      className="search"
      onChange={[Function]}
      placeholder="Search documents..."
      spellCheck={false}
      type="search"
    />
  </div>
  <div
    className="uploadControl"
  >
    <input
      accept="image/x-png,image/jpeg"
      id="upload"
      onChange={[MockFunction]}
      type="file"
    />
    <button
      className="button "
    >
      upload
    </button>
  </div>
</div>
`;
